lib/iris/_lazy_data.py

```diff
@@ -369,10 +369,15 @@ def map_complete_blocks(src, func, dims, out_sizes):
         Output size of dimensions that cannot be chunked.
 
     """
-    if not src.has_lazy_data():
+    if is_lazy_data(src):
+        data = src
+    elif not hasattr(src, "has_lazy_data"):
```

Comment:
Not sure whether this is the best way to check for a plain array.  I know that previously explicitly checking types has been discouraged (hence just checking for `compute` attribute in `_is_lazy_data`).

Reply:
I think I like it as a test because you're testing for the exact thing that will cause an attribute error if you let it through (because the next check is `src.has_lazy_data()`)?